Makefile

```diff
@@ -31,9 +31,9 @@ test: unit-tests build-examples
 unit-tests: $(tests_binary)
 	$^ --exclude=integration --sequential
 
-$(tests_binary): $(SOURCE_FILES) | $(BUILD_DIR)
+$(tests_binary): $(wildcard test/*.pony) $(SOURCE_FILES) | $(BUILD_DIR)
 	${GET_DEPENDENCIES_WIITH}
-	$(PONYC) -o ${BUILD_DIR} $(SRC_DIR)
+	$(PONYC) -o ${BUILD_DIR} -b $(PACKAGE) test
```

Comment:
With the above suggested change, I think this could be left the same as well. Assuming we are ok with the above change.

If we are going to do tests in the top level for libraries, I'd prefer to do it across all of them.